search/searcher/search_contains_test.go

```diff
@@ -0,0 +1,1040 @@
+	tests := []struct {
+		QueryShape       [][]float64
+		DocShapeVertices [][][]float64
+		DocShapeName     string
+		Desc             string
+		Expected         []string
+		QueryType        string
+	}{
+		{
+			QueryShape:       [][]float64{{1, 2}, {3, 5}},
+			DocShapeVertices: [][][]float64{{{1, 2}, {3, 5}, {2, 7}, {1, 2}}},
+			DocShapeName:     "polygon1",
+			Desc:             "linestring coinciding with edge of the polygon",
+			Expected:         []string{"polygon1"},
+			QueryType:        "contains",
+		},
+		{
+			QueryShape:       [][]float64{{1, 0}, {0, 1}},
+			DocShapeVertices: rightRect,
+			DocShapeName:     "polygon1",
+			Desc:             "diagonal of a square",
+			Expected:         []string{"polygon1"},
+			QueryType:        "contains",
+		},
+		{
+			QueryShape:       [][]float64{{0.2, 0.2}, {0.8, 0.8}},
+			DocShapeVertices: rightRect,
+			DocShapeName:     "polygon1",
+			Desc:             "linestring within polygon",
+			Expected:         []string{"polygon1"},
+			QueryType:        "contains",
+		},
+	}
+
+	i := setupIndex(t)
+
+	for _, test := range tests {
+		doc := document.NewDocument(test.DocShapeName)
+		doc.AddField(document.NewGeoShapeFieldWithIndexingOptions("geometry", []uint64{},
+			[][][][]float64{test.DocShapeVertices}, "polygon", document.DefaultGeoShapeIndexingOptions))
+		err := i.Update(doc)
+		if err != nil {
+			t.Error(err)
+		}
+
+		indexReader, err := i.Reader()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		t.Run(test.Desc, func(t *testing.T) {
+			got, err := runGeoShapeLinestringIntersectsQuery(test.QueryType,
+				indexReader, test.QueryShape, "geometry")
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !reflect.DeepEqual(got, test.Expected) {
+				t.Errorf("expected %v, got %v for linestring: %+v",
+					test.Expected, got, test.QueryShape)
+			}
+		})
+		err = i.Delete(doc.ID())
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+		err = indexReader.Close()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func TestEnvelopePointContains(t *testing.T) {
+	tests := []struct {
+		QueryShape       [][]float64
+		DocShapeVertices []float64
+		DocShapeName     string
+		Desc             string
+		Expected         []string
+		QueryType        string
+	}{
+		{
+			QueryShape:       [][]float64{{0, 1}, {1, 0}},
+			DocShapeVertices: []float64{0.5, 0.5},
+			DocShapeName:     "point1",
+			Desc:             "point completely within bounded rectangle",
+			Expected:         nil, // will always be nil since point can't contain envelope
+			QueryType:        "contains",
+		},
+	}
+
+	i := setupIndex(t)
+
+	for _, test := range tests {
+		doc := document.NewDocument(test.DocShapeName)
+		doc.AddField(document.NewGeoShapeFieldWithIndexingOptions("geometry", []uint64{},
+			[][][][]float64{{{test.DocShapeVertices}}}, "point", document.DefaultGeoShapeIndexingOptions))
+		err := i.Update(doc)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		indexReader, err := i.Reader()
```

Comment:
closing the index reader is till needed? and the same applies to many other test cases?